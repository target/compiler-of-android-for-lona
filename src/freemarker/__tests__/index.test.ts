import { parse, evaluate, ProgramNode } from '../index'

describe('Freemarker', () => {
  it('parses a template', () => {
    const template = '<#if x>yes</#if>'

    expect(parse(template).ast).toMatchSnapshot()
  })

  it('evaluates a template', () => {
    const template = '<#if x>yes</#if>'
    const ast = parse(template).ast as ProgramNode

    expect(evaluate(ast, { data: { x: true } })).toEqual('yes')
    expect(evaluate(ast, { data: { x: false } })).toEqual('')
  })
})
