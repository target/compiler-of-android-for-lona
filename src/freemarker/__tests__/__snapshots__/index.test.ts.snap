// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Freemarker parses a template 1`] = `
ProgramNode {
  "body": Array [
    ConditionNode {
      "consequent": Array [
        TextNode {
          "end": 10,
          "loc": Object {
            "end": Object {
              "column": 11,
              "line": 1,
            },
            "start": Object {
              "column": 8,
              "line": 1,
            },
          },
          "start": 7,
          "text": "yes",
          "type": "Text",
        },
      ],
      "end": 7,
      "loc": Object {
        "end": Object {
          "column": 8,
          "line": 1,
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "params": Object {
        "name": "x",
        "type": "Identifier",
      },
      "start": 0,
      "type": "Condition",
    },
  ],
  "end": 15,
  "loc": Object {
    "end": Object {
      "column": 16,
      "line": 1,
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "start": 0,
  "type": "Program",
}
`;
